src/walk_dirs.rs

```diff
@@ -144,13 +155,120 @@ pub fn discover_fences_and_files(start_path: &str) -> Vec<WalkFileData> {
     .collect();
 }
 
+
+fn get_imports_map(imports: &Vec<SourceSpecifiers>, file_path: &PathBuf) -> HashMap<String, Option<HashSet<String>>> {
+    let mut imports_map : HashMap<String, Option<HashSet<String>>> = HashMap::new();
+    // imports.iter().for_each(|import| {
+    //   import.specifiers
+    // });
+    imports.iter().for_each(|import| {
+    
+      let mut set: HashSet<String> = import.specifiers.iter().filter_map(|spec| -> Option<String> {
+        let file_text = std::fs::read(file_path).expect(&format!(
+          "error opening source file \"{:?}\"",
+          file_path
+        ));
+        if let Some(default) = spec.as_default() {
+          let text =  &file_text[default.span.lo().to_usize()-1..default.span.hi().to_usize()-1];
+          return Some(String::from_utf8_lossy(text).to_string());
+        }
+        if let Some(named) = spec.as_named() {
+          
+          let text =  &file_text[named.span.lo().to_usize()-1..named.span.hi().to_usize()-1];
+          println!("{}", String::from_utf8_lossy(text));
+          return Some(String::from_utf8_lossy(text).to_string());
+        }
+        None
+      }).collect();
+
+      if let Some(current_set) = imports_map.get(&import.source.value.to_string()) {
+        if let  Some(current_set) = current_set {
+          let mut new_set: HashSet<String> = HashSet::from_iter(current_set.iter().map(|v| v).cloned());
+          for val in set {
+            new_set.insert(val);
+          }
+          imports_map.insert(import.source.value.to_string(), Some(new_set.to_owned()));
```

Comment:
you should be able to mutate this instead of copying and overwriting, I think you need the map to be 
`HashMap<String, Option<mut HashSet<String>>>`?

Reply:
No worries if you grind your head against it too long and get stuck, we can always leave a TODO and file a github issue then revisit later

Reply:
Added a TODO comment on this